From: Matt Fleming <matt.fleming@intel.com>
Date: Tue, 28 Feb 2012 13:37:22 +0000
Subject: [PATCH 13/17] x86, mkpiggy: Don't open code put_unaligned_le32()

commit 12871c568305a0b20f116315479a18cd46882e9b upstream.

Use the new headers in tools/include instead of rolling our own
put_unaligned_le32() implementation.

Cc: H. Peter Anvin <hpa@zytor.com>
Signed-off-by: Matt Fleming <matt.fleming@intel.com>
Link: http://lkml.kernel.org/r/1330436245-24875-4-git-send-email-matt@console-pimps.org
Signed-off-by: H. Peter Anvin <hpa@zytor.com>
---
 arch/x86/boot/compressed/Makefile  |    1 +
 arch/x86/boot/compressed/mkpiggy.c |   11 ++---------
 2 files changed, 3 insertions(+), 9 deletions(-)

Index: linux-3.2.46/arch/x86/boot/compressed/Makefile
===================================================================
--- linux-3.2.46.orig/arch/x86/boot/compressed/Makefile	2013-07-26 19:19:52.000000000 +0000
+++ linux-3.2.46/arch/x86/boot/compressed/Makefile	2013-07-26 19:19:58.000000000 +0000
@@ -22,6 +22,7 @@
 LDFLAGS_vmlinux := -T
 
 hostprogs-y	:= mkpiggy
+HOST_EXTRACFLAGS += -I$(srctree)/tools/include
 
 VMLINUX_OBJS = $(obj)/vmlinux.lds $(obj)/head_$(BITS).o $(obj)/misc.o \
 	$(obj)/string.o $(obj)/cmdline.o $(obj)/early_serial_console.o \
Index: linux-3.2.46/arch/x86/boot/compressed/mkpiggy.c
===================================================================
--- linux-3.2.46.orig/arch/x86/boot/compressed/mkpiggy.c	2013-07-26 18:55:43.000000000 +0000
+++ linux-3.2.46/arch/x86/boot/compressed/mkpiggy.c	2013-07-26 19:19:58.000000000 +0000
@@ -29,14 +29,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <inttypes.h>
-
-static uint32_t getle32(const void *p)
-{
-	const uint8_t *cp = p;
-
-	return (uint32_t)cp[0] + ((uint32_t)cp[1] << 8) +
-		((uint32_t)cp[2] << 16) + ((uint32_t)cp[3] << 24);
-}
+#include <tools/le_byteshift.h>
 
 int main(int argc, char *argv[])
 {
@@ -69,7 +62,7 @@
 	}
 
 	ilen = ftell(f);
-	olen = getle32(&olen);
+	olen = get_unaligned_le32(&olen);
 	fclose(f);
 
 	/*
